Src/Support/Google.Apis.Auth/OAuth2/ServiceAccountCredential.cs

```diff
@@ -330,7 +355,9 @@ private Task<string> GetOrCreateJwtAccessTokenAsync(string authUri)
                     _jwtCache = new Dictionary<string, LinkedListNode<JwtCacheEntry>>();
                     _jwts = new LinkedList<JwtCacheEntry>();
                 }
-                if (_jwtCache.TryGetValue(authUri, out var cachedJwtNode))
+
+                string jwtKey = HasExplicitScopes ? ScopedTokenCacheKey : authUri;
```

Comment:
Instead of doing this here, I would pass a `string jwtKey` to this method and figure out which key is needed on the calling method, whic is the one that determines what is being generated and why.
I'd like to keep the cache logic separate from what we are caching and why we are caching what we are caching.

Reply:
makes sense